app/assets/javascripts/discourse/controllers/user-activity-bookmarks-with-reminders.js.es6

```diff
@@ -0,0 +1,54 @@
+import Controller from "@ember/controller";
+import { inject } from "@ember/controller";
+import discourseComputed from "discourse-common/utils/decorators";
+import Bookmark from "discourse/models/bookmark";
+
+export default Controller.extend({
+  application: inject(),
+  user: inject(),
+
+  content: null,
+  loading: false,
+  noResultsHelp: null,
+
+  loadItems() {
+    this.setProperties({
+      content: [],
+      loading: true,
+      noResultsHelp: null
+    });
+
+    this.model
+      .loadItems()
+      .then(response => {
+        if (response && response.no_results_help) {
+          this.set("noResultsHelp", response.no_results_help);
+        }
+
+        if (response && response.bookmarks) {
+          let bookmarks = [];
+          response.bookmarks.forEach(bookmark => {
+            bookmarks.push(Bookmark.create(bookmark));
+          });
+          this.content.pushObjects(bookmarks);
+        }
+      })
+      .finally(() =>
+        this.setProperties({
+          loaded: true,
+          loading: false
+        })
+      );
+  },
+
+  @discourseComputed("loaded", "content.[]")
+  noContent(loaded, content) {
+    return loaded && content.length === 0;
+  },
+
+  actions: {
+    removeBookmark(bookmark) {
+      return bookmark.destroy().then(() => this.loadItems());
```

Comment:
I think this should work:

```suggestion
      return bookmark.destroy().then(this.loadItems);
```

Reply:
For the record this did not work, because inside loadItems `this` is pointing to the wrong thing.